rpc-extractor: add uptime and getnettotals support

GuiSchet opened the pull request:
Implements two new RPC queries to the rpc-extractor:
  - uptime: Returns node uptime in seconds
  - getnettotals: Returns network traffic statistics (bytes sent/received, upload target)

  Both RPCs follow the established pattern with:
  - Protobuf message definitions in rpc.proto
  - From trait implementations for type conversions
  - CLI flags --disable-uptime and --disable-getnettotals
  - Async functions that publish events to NATS

  The uptime RPC is particularly lightweight (4-5ms response time) and
  provides a simple metric for node health monitoring. The getnettotals
  RPC exposes network traffic statistics useful for bandwidth analysis.

  Partially addresses #199.

0xB10C commented on protobuf/rpc.proto:
```diff
@@ -78,3 +80,21 @@ message MempoolInfo {
+  required bool   serve_historical_blocks  = 4; // True if serving historical blocks
+  required uint64 bytes_left_in_cycle      = 5; // Bytes left in current time cycle
+  required uint64 time_left_in_cycle       = 6; // Seconds left in current time cycle
+}
```
Nice! However, I'm not sure if we are going to use `-maxuploadtarget=` much in the monitoring nodes. I guess it's only minimal bandwidth overhead, so it doesn't hurt much.

Not sure if metrics for it make sense.

GuiSchet replied:
Good point. The overhead is negligible, so we can likely skip the metrics.

0xB10C commented:
feel free to just add a match arm similar to:
```rust
 &shared::protobuf::rpc::rpc_event::Event::Uptime(_) | &shared::protobuf::rpc::rpc_event::Event::NetTotals(_) => {},
```
in metrics for now.


As for metrics, uptime should probably just be a prometheus IntGauge. For `total_bytes_received` and `total_bytes_sent`, am IntGauge works too.

GuiSchet commented:
Hi @0xB10C,

Apologies for the CI spam! I've now addressed all the issues:

### Changes Made:

1. **Prometheus Metrics Added**:
     - `rpc_uptime` (IntGauge) - Node uptime in seconds
     - `rpc_nettotals_total_bytes_received` (IntGauge) - Total bytes received
     - `rpc_nettotals_total_bytes_sent` (IntGauge) - Total bytes sent

2. **Integration Tests Added**:
     - `test_integration_rpc_uptime()` - Verifies uptime RPC events
     - `test_integration_rpc_getnettotals()` - Verifies getnettotals RPC events
     - Updated existing test helpers to support the new `--disable-uptime` and `--disable-getnettotals` flags

3. **Fixed**:
     - Non-exhaustive pattern match error in metrics tool
     - All integration tests now pass locally (verified with `cargo test --features nats_integration_tests --features
  node_integration_tests`)

  ### Test Results:
  ✅ test_integration_rpc_getmempoolinfo ... ok
  ✅ test_integration_rpc_getnettotals ... ok
  ✅ test_integration_rpc_getpeerinfo ... ok
  ✅ test_integration_rpc_uptime ... ok

  All 40 metrics integration tests also pass.

  Ready for another look when you have time. Thanks for the feedback!
